File: app/src/main/java/com/vivekpanchal/newshub/database/NewsHeadlineDao.java
package com.vivekpanchal.newshub.database;

import android.arch.lifecycle.LiveData;
import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;
import android.arch.persistence.room.Update;

import java.util.List;

@Dao
public interface NewsHeadlineDao {

    @Query("SELECT * FROM news_headlines ORDER BY id")
    LiveData<List<NewsHeadlineEntity>> loadAllNewsHeadlines();

    @Insert
    void insertNewsHeadline(NewsHeadlineEntity headlineEntity);

    @Update(onConflict = OnConflictStrategy.REPLACE)
    void updateNewsHeadline(NewsHeadlineEntity headlineEntity);

    @Delete
    void deleteNewsHeadline(NewsHeadlineEntity headlineEntity);

    @Query("SELECT * FROM news_headlines WHERE news_headline = :headline")
    LiveData<NewsHeadlineEntity> loadNewsByHeadline(String headline);
}

File: app/src/main/java/com/vivekpanchal/newshub/data/local/NewsHeadlineDao.kt
package com.vivekpanchal.newshub.data.local

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import kotlinx.coroutines.flow.Flow

@Dao
interface NewsHeadlineDao {

    @Query("SELECT * FROM news_headlines ORDER BY id DESC")
    fun loadAllNewsHeadlines(): Flow<List<NewsHeadlineEntity>>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertNewsHeadline(headlineEntity: NewsHeadlineEntity)

    @Delete
    suspend fun deleteNewsHeadline(headlineEntity: NewsHeadlineEntity)

    @Query("SELECT * FROM news_headlines WHERE news_headline = :headline LIMIT 1")
    fun loadNewsByHeadline(headline: String): Flow<NewsHeadlineEntity?>
}
